tests/unit/core/client/test_ConnectionManager.py

```diff
@@ -161,28 +162,74 @@ def tearDown(self):
     def test_success(self):
         # test to see that if we setup a server, we get a connection
         self.server = MocServerProtocolFactory(self.clock)
-        self.server_port = reactor.listenTCP(PEER_PORT,self.server,interface=LOCAL_HOST)
-
-        d = yield self.connection_manager._manage()
+        self.server_port = reactor.listenTCP(PEER_PORT, self.server, interface=LOCAL_HOST)
+        d = yield self.connection_manager.manage(schedule_next_call=False)
+        self.assertEqual(1, self.connection_manager.num_peer_connections())
+        d = yield self.connection_manager._peer_connections[self.TEST_PEER].factory.connection_was_made_deferred
+        self.assertEqual(0, self.connection_manager.num_peer_connections())
+        self.assertEqual(True, d)
         self.assertEqual(1, self.TEST_PEER.success_count)
-
+        self.assertEqual(0, self.TEST_PEER.down_count)
 
     @defer.inlineCallbacks
     def test_bad_server(self):
         # test to see that if we setup a server that returns an improper reply
         # we don't get a connection
         self.server = MocServerProtocolFactory(self.clock, is_good=False)
-        self.server_port = reactor.listenTCP(PEER_PORT,self.server,interface=LOCAL_HOST)
-
-        d = yield self.connection_manager._manage()
+        self.server_port = reactor.listenTCP(PEER_PORT, self.server, interface=LOCAL_HOST)
+        d = yield self.connection_manager.manage(schedule_next_call=False)
+        self.assertEqual(1, self.connection_manager.num_peer_connections())
+        d = yield self.connection_manager._peer_connections[self.TEST_PEER].factory.connection_was_made_deferred
+        self.assertEqual(0, self.connection_manager.num_peer_connections())
+        self.assertEqual(True, d)
         self.assertEqual(0, self.TEST_PEER.success_count)
         self.assertEqual(1, self.TEST_PEER.down_count)
 
     @defer.inlineCallbacks
     def test_non_existing_server(self):
         # Test to see that if we don't setup a server, we don't get a connection
-        d = yield self.connection_manager._manage()
+        d = yield self.connection_manager.manage(schedule_next_call=False)
+        self.assertEqual(1, self.connection_manager.num_peer_connections())
+        d = yield self.connection_manager._peer_connections[self.TEST_PEER].factory.connection_was_made_deferred
+        self.assertEqual(0, self.connection_manager.num_peer_connections())
+        self.assertEqual(False, d)
+        self.assertEqual(0, self.connection_manager.num_peer_connections())
+        self.assertEqual(0, self.TEST_PEER.success_count)
+        self.assertEqual(1, self.TEST_PEER.down_count)
+
```

Comment:
Added tests to see if multiple parallel connections can be started using the manage() function